daemon/src/lib.rs

```diff
@@ -118,9 +168,13 @@ impl TakerListenProtocols {
             (self.offers, offers.into()),
         ]
     }
+
+    pub fn compatible_maker_protocols() -> HashSet<String> {
+        MakerProtocols::default().supported_protocols()
+    }
```

Comment:
📣 As soon as the maker starts supporting more than one version of a `libp2p`-based protocol for backwards compatibility (for example, after this https://github.com/itchysats/itchysats/pull/2378 gets merged), this won't work.


For example:

- Taker supports `/foo/2.0.0`.
- Build-time maker supports `/foo/1.0.0` and `/foo/2.0.0`, which is the value we get here.
- Run-time maker no longer supports `/foo/1.0.0` for whatever reason.
- Taker connects with run-time maker. We display an error even though the taker doesn't care about (cannot even perform) `/foo/1.0.0`.

Reply:
> For example:
> 
>     * Taker supports `/foo/2.0.0`.
> 
>     * Build-time maker supports `/foo/1.0.0` and `/foo/2.0.0`, which is the value we get here.
> 
>     * Run-time maker no longer supports `/foo/1.0.0` for whatever reason.
> 
>     * Taker connects with run-time maker. We display an error even though the taker doesn't care about (cannot even perform) `/foo/1.0.0`.

I think you are right, we have to be explicit about what the taker expects on the other side. This sucks a bit though, because we have to manually ensure that it is up to date :/

Reply:
I changed #2161 to introduce a `ProtocolFactory` and explicitly use the latest maker protocols for what the taker expects. I think this should solve the problem.